skimage/morphology/grey.py

```diff
@@ -62,16 +130,15 @@ def erosion(image, selem=None, out=None, shift_x=False, shift_y=False):
            [0, 0, 0, 0, 0]], dtype=uint8)
 
     """
-
-    if image is out:
-        raise NotImplementedError("In-place erosion not supported!")
-    image = img_as_ubyte(image)
-    selem = img_as_ubyte(selem)
-    return cmorph._erode(image, selem, out=out,
-                         shift_x=shift_x, shift_y=shift_y)
+    selem = np.array(selem)
+    selem = _shift_selem(selem, shift_x, shift_y)
+    if out is None:
+        out = np.empty_like(image)
+    nd.grey_erosion(image, footprint=selem, output=out)
+    return out
 
 
-@default_fallback
+@default_selem
 def dilation(image, selem=None, out=None, shift_x=False, shift_y=False):
```

Comment:
I'm just going to toss this out there.

Extrapolating from @jni's example above, I don't think the eccentricity of any structuring element matters! The result should be equivalent if it's padded asymmetrically with zeros until all axes have odd numbers of elements, so there is a true center, and then used. I can't think of any case where this would be false.

However, if I'm wrong and eccentricity actually does matter in some obscure cases, a better and more intuitive API than `shift_*` would be to explicitly define the center point of the selem. Accept a tuple `center` and use that. Computation would then be more efficient than padding with zeros, and it would be fully general (if needed!)